spec_util/meld.go

```diff
@@ -117,11 +117,25 @@ func mergeExampleValues(dst, src *pb.Data) {
 		// Sort keys to ensure a stable selection.
 		sort.Strings(keys)
```

Comment:
Instead of doing 2 loops, how about we just collect all the keys from src and dst into 2 arrays, sort both, then take the first element from each?

Reply:
That's basically what the old code did.  The problem is that if we're merging, say, `["ex1", "ex2"]` and `["ex3", "ex4"]`, we want `["ex1", "ex2"]`, not `["ex1", "ex3"]`.

I did it this way to avoid looping over all the elements of each example map.  On the other hand, it should be small unless a user uploads a spec with a lot of examples, so a simpler approach might be to add all examples from `src` and `dst` to `examples`, sort the keys, and remove all but the smallest two.  Thoughts?

Reply:
> I did it this way to avoid looping over all the elements of each example map. On the other hand, it should be small unless a user uploads a spec with a lot of examples, so a simpler approach might be to add all examples from src and dst to examples, sort the keys, and remove all but the smallest two. Thoughts?

Yeah let's do this. I don't think you even need to add all examples together, just get all the keys, get the smallest 2, and find them in either `src` or `dst`

Reply:
Sounds good to me.  Done.